pkg/roachprod/install/cluster_synced.go

```diff
@@ -2654,6 +2654,33 @@ func (c *SyncedCluster) pghosts(
 	return m, nil
 }
 
+// resolveLoadBalancerURL resolves the load balancer postgres URL for the given
+// virtual cluster and SQL instance. Returns an empty string if a load balancer
+// is not found.
+func (c *SyncedCluster) loadBalancerURL(
+	ctx context.Context, l *logger.Logger, virtualClusterName string, sqlInstance int,
+) (string, error) {
+	services, err := c.DiscoverServices(ctx, virtualClusterName, ServiceTypeSQL)
+	if err != nil {
+		return "", err
+	}
+	port := 0
+	serviceMode := ServiceModeShared
+	for _, service := range services {
+		if service.VirtualClusterName == virtualClusterName && service.Instance == sqlInstance {
+			serviceMode = service.ServiceMode
+			port = service.Port
+			break
+		}
+	}
+	address, err := c.FindLoadBalancer(l, port)
+	if err != nil || address == nil {
+		return "", err
+	}
+	loadBalancerURL := c.NodeURL(address.IP, address.Port, virtualClusterName, serviceMode, AuthRootCert)
```

Comment:
Should we take `auth` as parameter? I know it will complicate matters for the expanders, but assuming root doesn't seem right either.

Reply:
Nice catch, yes we should pass `auth`. Done. Also updated to use the same auth as the other expanders.